examples/EF_QAE/README.md

```diff
@@ -0,0 +1,76 @@
+
+A graphical depiction of a quantum encoder can be seen in the following figure. In a quantum encoder the information contained in some of the input qubits must be discarded after the initial encoding. Then, fresh qubits (here initialized to the |0> state, but one may consider any other easy-to-construct reference state) are prepared and used to implement the final decoding, which is finally compared to the initial state.
+
+![autoencoder](images/autoencoder.png)
+
+In the following, however, we present the EF-QAE. A schematic diagram of the EF-QAE can be seen in the next figure. The algorithm can be initialized with a set of initial states, a feature vector **x**, and a shallow sequence of quantum gates U. In this scheme, we define a unitary U(**θ**, **x**) acting on the initial state, where **x** is a feature vector that characterizes the set of input states. For instance, as we will see, **x** may be the transverse field λ of the 1D Ising spin chain. Once the trial state is prepared, measurements are performed to evaluate the cost function C(**θ**). This result is then fed into the classical optimizer, where the parameters **θ** are adjusted in a quantum-classical loop until the cost function converges to a value close to 0. When the loop terminates, U(**θ**<sub>opt</sub>, **x**) prepares compressed states of a particular model.
+
+![diagram](images/diagram.png)
+
+A figure of merit for the wrong answer when training is simply the total amount of non-zero measurement outcomes on the discarded qubits, which shall be minimized. In order to design the cost function to be local, different outcomes may be penalized by their Hamming distance to the |0> state, which is just the number of symbols that are different in the binary representation. Notice that this cost function has a value of zero if and only if the compression is successfully completed.
+
+To implement the quantum autoencoder model on a quantum computer we must define the form of the parametrized unitary, decomposing it into a quantum circuit suitable for optimization. This can be done using a similar structure to that in the following figure.
+
+![ansatz](images/ansatz.png)
+
+We now encode the feature vector **x** into each of the single R<sub>y</sub> qubit rotations by using a linear function as
+
+![Ry](images/Ry.png)
+
+where *i*,*j* simply indicates a component of the vector, and **θ** are the parameters adjusted in the optimization loop.
+
+## How to run an example?
+
+To run a particular instance of the problem we have to set up the initial
+arguments:
+- `layers` (int): number of ansatz layers.
+- `autoencoder` (int): 0 to run the EF-QAE or 1 to run the QAE.
+- `example` (int): 0 to run Ising model example or 1 to run the Handwritten digits example.
```

Comment:
Could you please report this info in the main function, through argparse using the `help` option?